ipa-core/src/protocol/ipa_prf/boolean_ops/addition_sequential.rs

```diff
@@ -254,9 +254,16 @@ mod test {
                     .await
                     .unwrap()
                 })
-                .await
+                .await;
+
+            let result_refs: [&BitDecomposed<_>; 3] =
+                result_bits.iter().collect::<Vec<_>>().try_into().unwrap();
+
+            let result = result_refs
```

Comment:
I had to do the same thing. It's annoying that `Reconstruct` is only implemented for `[&BitDecomposed<_>; 3]` and not for `[BitDecomposed<_>; 3]`. Is there any way we can add support for `Reconstruct` to the output of `world.upgraded_semi_honest()` so that we can just call `.reconstruct()` on it directly?